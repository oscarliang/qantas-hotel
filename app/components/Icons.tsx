import React from 'react';

interface IconProps {
    filled: boolean;
}

export const StarIcon: React.FC<IconProps> = ({ filled }) => {
    return (
        <svg
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 24 24"
            width="16"
            height="16"
            fill={filled ? "#fef08a" : "none"}
            stroke="#fef08a"
            strokeWidth="1"
            className="inline-block mx-0.5"
        >
            <path d="M12 2l3.09 6.26L22 9.27l-5 4.87 1.18 6.88L12 17.77l-6.18 3.25L7 14.14 2 9.27l6.91-1.01L12 2z" />
        </svg>
    );
};

export const CircleIcon: React.FC<IconProps> = ({ filled }) => {
    return (
        <svg
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 24 24"
            width="16"
            height="16"
            fill={filled ? "#fef08a" : "none"}
            stroke="#fef08a"
            strokeWidth="1"
            className="inline-block mx-0.5"
        >
            <circle cx="12" cy="12" r="8" />
        </svg>
    );
}; 