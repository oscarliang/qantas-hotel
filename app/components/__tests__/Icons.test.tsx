import React from 'react';
import { render } from '@testing-library/react';
import { StarIcon, CircleIcon } from '../Icons';

describe('Icons Component', () => {
  describe('StarIcon', () => {
    it('renders filled star correctly', () => {
      const { container } = render(<StarIcon filled={true} />);
      const svg = container.querySelector('svg');
      expect(svg).toBeInTheDocument();
      expect(svg).toHaveAttribute('fill');
      expect(container).toMatchSnapshot();
    });

    it('renders empty star correctly', () => {
      const { container } = render(<StarIcon filled={false} />);
      const svg = container.querySelector('svg');
      expect(svg).toBeInTheDocument();
      expect(svg).toHaveAttribute('fill', 'none');
      expect(container).toMatchSnapshot();
    });
  });

  describe('CircleIcon', () => {
    it('renders filled circle correctly', () => {
      const { container } = render(<CircleIcon filled={true} />);
      const svg = container.querySelector('svg');
      expect(svg).toBeInTheDocument();
      expect(svg).toHaveAttribute('fill');
      expect(container).toMatchSnapshot();
    });

    it('renders empty circle correctly', () => {
      const { container } = render(<CircleIcon filled={false} />);
      const svg = container.querySelector('svg');
      expect(svg).toBeInTheDocument();
      expect(svg).toHaveAttribute('fill', 'none');
      expect(container).toMatchSnapshot();
    });
  });
});
