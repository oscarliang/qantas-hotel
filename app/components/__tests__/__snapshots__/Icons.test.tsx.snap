// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons Component CircleIcon renders empty circle correctly 1`] = `
<div>
  <svg
    class="inline-block mx-0.5"
    fill="none"
    height="16"
    stroke="#fef08a"
    stroke-width="1"
    viewBox="0 0 24 24"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="12"
      cy="12"
      r="8"
    />
  </svg>
</div>
`;

exports[`Icons Component CircleIcon renders filled circle correctly 1`] = `
<div>
  <svg
    class="inline-block mx-0.5"
    fill="#fef08a"
    height="16"
    stroke="#fef08a"
    stroke-width="1"
    viewBox="0 0 24 24"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="12"
      cy="12"
      r="8"
    />
  </svg>
</div>
`;

exports[`Icons Component StarIcon renders empty star correctly 1`] = `
<div>
  <svg
    class="inline-block mx-0.5"
    fill="none"
    height="16"
    stroke="#fef08a"
    stroke-width="1"
    viewBox="0 0 24 24"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12 2l3.09 6.26L22 9.27l-5 4.87 1.18 6.88L12 17.77l-6.18 3.25L7 14.14 2 9.27l6.91-1.01L12 2z"
    />
  </svg>
</div>
`;

exports[`Icons Component StarIcon renders filled star correctly 1`] = `
<div>
  <svg
    class="inline-block mx-0.5"
    fill="#fef08a"
    height="16"
    stroke="#fef08a"
    stroke-width="1"
    viewBox="0 0 24 24"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12 2l3.09 6.26L22 9.27l-5 4.87 1.18 6.88L12 17.77l-6.18 3.25L7 14.14 2 9.27l6.91-1.01L12 2z"
    />
  </svg>
</div>
`;
